### ISSUE TITLE
Foxy Proxy on Mullvad Browser doesnt proxy at all

### ISSUE THREAD

**Issue by NearestWater262** [foxyproxy/browser-extension#141]
Fresh Mullvad flatpak install (Linux)

Go to mozilla addons, install foxy

add basic socks5 with host and port

go to a site to see your IP address.

It doesnt proxy at all, just uses your regular IP

i am stuck using old version, 7.5.1, and for that version, I have to turn off the Mullvad DNS and just use system DNS resolver, and then it works, and I tried that on the newest version too, but still didnt work, same behavior.

**Comment by erosman** [comment]
Mullvad Browser seems to have built-in VPN/TOR settings that might interfere with the proxy settings.

Have you enabled Private Browsing Permission?
Are you using patterns on single proxy mode?

**Comment by NearestWater262** [comment]
Yes I enable Private Browsing permission.

No I did not use any patterns, just installed and went straight to adding a basic socks5 proxy with a title, host, and port.

Here is the network settings that are default in Mullvad browser below

![image](https://github.com/user-attachments/assets/11fc18ee-8088-47c4-bae5-578b5df4cfd6)

**Comment by erosman** [comment]
From Network Settings, FoxyProxy is not controlling the proxy settings.

Is the FoxyProxy icon on the toolbar in colour or grey?

Also...

- Open FoxyProxy Options page
- Open Developer Tools (F12)
- Go to Console tab
- Paste the following
```js
browser.proxy.settings.get({})
.then(console.log)
.catch(console.error)
```
- Enter

What does it say?
You can remove private data before posting in here.

**Comment by NearestWater262** [comment]
it's in colour

![image](https://github.com/user-attachments/assets/2e2791b2-501d-43e7-8d2c-1106a19c89f9)

**Comment by erosman** [comment]
Above shows that FoxyProxy is handling the proxying but it is set to the system i.e. the mode is "Disable".
What happens when you change the mode via the toolbar popup?
Does the icon change?

**Comment by NearestWater262** [comment]
> What happens when you change the mode via the toolbar popup?

what mode? what toolbar popup? please explain with a screenshot? Sorry, just confused and want to make sure I know exactly what you mean.

@erosman

**Comment by NearestWater262** [comment]
also, suddenly, this is appearing below but still is not proxying, still using my regular IP

![image](https://github.com/user-attachments/assets/5841880c-31e8-4df0-80d0-3167bf3235b7)

**Comment by erosman** [comment]
> what mode? what toolbar popup? please explain with a screenshot?

In FoxyProxy, user add. edit their proxies on the Options page.
Selecting which proxy to use, or disable proxy (i.e. the mode) is done via the toolbar icon popup.

![FoxyProxy Popup](https://raw.githubusercontent.com/foxyproxy/browser-extension/main/screenshots/8.1/light-theme/popup.jpg)

**Comment by NearestWater262** [comment]
> Above shows that FoxyProxy is handling the proxying but it is set to the system i.e. the mode is "Disable". What happens when you change the mode via the toolbar popup? Does the icon change?

![image](https://github.com/user-attachments/assets/2b3666e9-7f71-4e7c-a718-3e60986eafa7)
![image](https://github.com/user-attachments/assets/c5a79caf-1e55-40e1-874a-19afe25b7ee9)

hope this answers your question. maybe you can give Mullvad Browser a try for yourself

**Comment by erosman** [comment]
There seems to be an issue with browser identification and FoxyProxy sees Mullvand browser as Chrome which would break the functionality.
I will run some tests and get back to you.

**Comment by NearestWater262** [comment]
> There seems to be an issue with browser identification and FoxyProxy sees Mullvand browser as Chrome which would break the functionality. I will run some tests and get back to you.

Hmm.. very weird indeed. Thank you for looking into this. I am use 7.5.1 which works fine, maybe something in that version's code will help you.

**Comment by erosman** [comment]
FoxyProxy v7.5.1 was a Firefox only version. FoxyProxy v8+ has a unified code for both Firefox & Chrome.

**Comment by NearestWater262** [comment]
this isn't mullvad browser issue because older version of Foxy Proxy works correctly

any update?

**Comment by erosman** [comment]
Can the browser be run without installation (like Nightly)?
I downloaded it but didn't find any instruction on running it WITHOUT installing it.

_mullvad-browser-linux-x86_64-13.5.1.tar.xz
Ubuntu 24.04_

**Comment by erosman** [comment]
> @erosman yes, you can extract the tar and just execute the binary like with tor browser bundle, you can execute the `start-mullvad-browser.desktop`. just be sure you mark it executable with `sudo chmod +x start-mullvad-browser.desktop`

It doesn't run for me.

**Comment by NearestWater262** [comment]
@erosman not sure what to say without more info about the errors..
maybe try running the script under Browser/start-mullvad-browser

**Comment by NearestWater262** [comment]
you can also get it on Flathub, if you have Flatpak. Very easy to uninstall when you are done with it.

**Comment by erosman** [comment]
I got it running with the help from [Starting the Mullvad Browser](https://mullvad.net/en/help/install-mullvad-browser#linux-start).

Sadly, it wont allow unpacked or unsigned extension installation. 

However, I found the problem which is the browser detection.

In fact, the issue has already been fixed in v8.10 subsequent to another issue in [Proxy by patterns not working if firefox users change their userAgent to another platform](https://github.com/foxyproxy/firefox-extension/issues/220).

Please wait for v8.10 release or for now try using v7.5.1 from https://addons.mozilla.org/en-US/firefox/addon/foxyproxy-standard/versions/ .

**Comment by NearestWater262** [comment]
> Sadly, it wont allow unpacked or unsigned extension installation.
there is a config to allow them, `xpinstall.signatures.required` change to `false`

> However, I found the problem which is the browser detection.
> 
> In fact, the issue has already been fixed in v8.10 subsequent to another issue in [Proxy by patterns not working if firefox users change their userAgent to another platform](https://github.com/foxyproxy/firefox-extension/issues/220).
> 
> Please wait for v8.10 release or for now try using v7.5.1 from https://addons.mozilla.org/en-US/firefox/addon/foxyproxy-standard/versions/ .

okay, awesome! Is there an estimation of when the 8.10 release is?

**Comment by erosman** [comment]
> Is there an estimation of when the 8.10 release is?

I will work on preparing a release as soon as possible.

**Comment by harryqt** [comment]
Any update on this?

**Comment by erosman** [comment]
I can update the current beta to the repo. Can you use the beta?

**Comment by harryqt** [comment]
Yup

**Comment by erosman** [comment]
OK, I will upload the beta to the repo shortly.

Please let us know if there are any issues, as it is not fully tested.

**Comment by erosman** [comment]
It is uploaded to the repo.

**Comment by harryqt** [comment]
Working on Mullvad browser. 👍

<img width="1512" alt="Screenshot 2024-08-18 at 4 49 21 PM" src="https://github.com/user-attachments/assets/d73618de-6072-4e0a-80f3-7a2f6e8df429">

**Comment by erosman** [comment]
There are some changes to the pop UI & processes in v8.10.
Can you test the Tab Proxy?

**Comment by harryqt** [comment]
Tab Proxy working fine. 👍

**Comment by harryqt** [comment]
@erosman When do you plan to push the release to Firefox extension store?

**Comment by erosman** [comment]
We haven't fully tested the changes yet. Once it is done, it will be uploaded to the AMO.

**Comment by erosman** [comment]
I am sorry for the delay in releasing v8.10.

I am the only developer and work on the free to use extension during my spare time.
Users are quick to flood the repo, AMO, and Chrome Web-store with unkind (_and even  harsh_) comments if there is a bug but contribution to beta testing are hard to come by.

**Comment by NearestWater262** [comment]
Do you still need more testing? I can help test. just let me know exactly how to.

**Comment by NearestWater262** [comment]
@erosman sorry for the delay, I was busy with some things

I tested basic proxy functionality for socks5, and seems to work fine with no leaks. Even when ublock's "Uncloak canonical names" is enabled, it does not leak DNS, where before this was a long standing issue that would leak DNS if this ublock option was enabled. I don't know if this was a fix on your side, ublock's side, or mozilla's side.

I also tested HTTP proxy and this seemed to work as well, no DNS leaks that I could see

As for Tab Proxy, it does work, however there are a couple of bad UX oversights I think.

1, When you open a new blank tab, then you select the foxy proxy extension to open the pop-out view, then you select the Tab Proxy dropdown, then you select the proxy you would like, foxy does not actually change the proxy for that tab either visually on the icon or functionally when going to a site. It is only after first connecting to a website that you can successfully change the proxy of that tab. I believe that this renders this feature fairly worthless as the server can clearly see that you are connecting to the site with no proxy or with your default proxy, then switching the proxy/IP address. So it becomes worthless from a privacy standpoint. I don't know if this is something that is in your control but I thought i would point it out.

2, When changing the tab proxy while on a website, the little label on the foxyproxy icon will indeed show the name of the proxy you've changed to, but when clicking out of the pop-out and then opening the pop-out again, the Tab Proxy dropdown will be blank. I really think this is not good for UX. The user should be reminded that the reason why the default chosen proxy in their list and the proxy on the icon label are mismatched is because they chose a Tab Proxy, at least with the Tab Proxy dropdown selection being persisted.
See here, not persisted:
![image](https://github.com/user-attachments/assets/3741294f-890d-428b-9ceb-dc941ac148e4)

All of my testing was done in Mullvad Browser with latest uBlock Origin and using the main branch of foxyproxy

**Comment by erosman** [comment]
@NearestWater262 Thank you for testing. 

> When you open a new blank tab, ...

Added option to set Tab Proxy for a new tab (#157)

> When changing the tab proxy while on a website, ...

There was an oversight which is fixed now.

I am working on the release of v8.10

**Comment by NearestWater262** [comment]
It looks like this is broken again. I am using 8.10. Using a simple socks5 proxy, Mullvad does not proxy DNS at all even when the switch to proxy DNS is enabled in foxy. During a, IP leak test, it connects to both the socks 5 proxy and also leaks my real IP.

Should I re-open? or make a new issue.
